src/api/raw.rs

```diff
@@ -196,12 +196,86 @@ pub struct ListParams {
     pub timeout: Option<u32>
 }
 
-/// Common query parameters for put/post/patch calls
+/// Common query parameters for put/post calls
 #[derive(Default, Clone)]
 pub struct PostParams {
     pub dry_run: bool,
 }
 
+/// Common query parameters for patch calls
+#[derive(Default, Clone)]
+pub struct PatchParams {
+    pub dry_run: bool,
+    /// Strategy which will be used. Defaults to `PatchStrategy::Strategic`
+    pub patch_strategy: PatchStrategy,
+    /// force Apply requests. Applicable only to `PatchStrategy::Apply`
+    pub force: bool,
+    /// fieldManager is a name of the actor that is making changes. Required for `PatchStrategy::Apply`
+    /// optional for everything else
+    pub field_manager: Option<String>
+}
+
+impl PatchParams {
+    fn validate(&self) -> Result<()> {
+        if let Some(field_manager) = &self.field_manager {
+            // that's not complete, go reference code does list all Unicode printable characters as valid, 
+            // this validation is a bit stricter 
+            if field_manager.len() > 128 || !field_manager.chars().all(|f| f.is_alphanumeric() || f == ' ') {
+            return Err(ErrorKind::RequestValidation("Failed to validate PatchParameters::field_manager!".to_owned()).into())
```

Comment:
if k8s accepts more than what we validate here, i'm vary of including this check.

Reply:
I didn’t find an easy way to get unicode category for ‘char’ in rust, so I cannot implement https://golang.org/pkg/unicode/#IsPrint. And I’m not sure that there is a crate for that.
I can remove chars validation completely, though. API server will validate it anyway and return an error as expected

Reply:
Given it'll be rejected upstream when it's invalid, I think it's fine to skip this. It's probably possible with a `regex` solution, [since IsPrint is "just" two range checks](https://golang.org/src/strconv/makeisprint.go), but it feels like overkill when it's already checked upstream, and it's almost certainly a programmer error if you encounter it.

Reply:
Removed the letter-checking bit, `len` validation is still kept in place, because i think that the fast request is the one you didn't make :)